src/inmanta/server/services/notificationservice.py

```diff
@@ -64,14 +64,28 @@ async def _cleanup(self) -> None:
         await data.Notification.clean_up_notifications()
 
     async def compile_done(self, compile: data.Compile) -> None:
-        if not compile.success and compile.do_export:
-            await self.notify(
-                compile.environment,
-                title="Compilation failed",
-                message="An exporting compile has failed",
-                severity=const.NotificationSeverity.error,
-                uri=f"/api/v2/compilereport/{compile.id}",
-            )
+        if not compile.success:
+            if compile.do_export:
+                await self.notify(
+                    compile.environment,
+                    title="Compilation failed",
+                    message="An exporting compile has failed",
+                    severity=const.NotificationSeverity.error,
+                    uri=f"/api/v2/compilereport/{compile.id}",
+                )
+            else:
+                reports = (await data.Compile.get_report(compile_id=compile.id))["reports"]
+                failed_pull_stage = next(
+                    (report for report in reports if report["name"] == "Pulling updates" and len(report["errstream"]) > 0), None
```

Comment:
Shouldn't we test the `returncode` instead. I guess we also want to raise a notification when a failed pull doesn't have anything in `errstream`.